detox/.detoxrc.json

```
@@ -13,7 +13,7 @@
     "ios.release": {
       "type": "ios.app",
       "binaryPath": "../ios/Build/Products/Release-iphonesimulator/Mattermost.app",
-      "build": "cd ../fastlane && NODE_ENV=production bundle exec fastlane ios simulator && cd ../detox"
+      "build": "cd .. && npm run build:ios-sim && cd detox"
```

By doing it like this you are forcing the script to clean up and install dependencies again, is that what you want?

To avoid that step do add an environment variable to skip setup like this

```suggestion
      "build": "cd .. && SKIP_SETUP=1 npm run build:ios-sim && cd detox"
```

---

Thanks and yes, that's what I wanted! This is release build for E2E. In local, it makes it convenient so contributor don't have to deal with dev [setup](https://github.com/mattermost/mattermost-mobile/blob/main/scripts/build.sh#L40-L81). But more convenient in CI so we could take advantage of using such setup instead of [dealing with it individually](https://git.internal.mattermost.com/qa/mobile-e2e-testing/-/blob/main/scripts/ios-build-pr.sh#L23-38).

I'm fine either way and I'll let @metanerd or @josephbaylon to decide which is better.

---

Opss sorry, what I meant was I prefer my original change where it didn't have `SKIP_SETUP`. It makes it easier in local. In CI, it could just be:
```bash
export NODE_OPTIONS=-"-max_old_space_size=12000"

cd mattermost-mobile/detox
NODE_ENV=development npm ci --cache .npm --prefer-offline
npm run e2e:ios-build-release
```

However, it will not help with cache (see [setup](https://github.com/mattermost/mattermost-mobile/blob/main/scripts/build.sh#L40-L81)) like what you have with [npm ci](https://git.internal.mattermost.com/qa/mobile-e2e-testing/-/blob/main/scripts/ios-build-pr.sh#L26) or [path to bundle install](https://git.internal.mattermost.com/qa/mobile-e2e-testing/-/blob/main/scripts/ios-build-pr.sh#L37). So, you might prefer to have `SKIP_SETUP=1` here and remain the same with what we have currently in [CI script](https://git.internal.mattermost.com/qa/mobile-e2e-testing/-/blob/main/scripts/ios-build-pr.sh#L23-42).

re: "dealing with it individually" is about doing [each step like what you have currently](https://git.internal.mattermost.com/qa/mobile-e2e-testing/-/blob/main/scripts/ios-build-pr.sh#L26-37): install dependency, patch package, install ios gems, install pod, generate assets and install fastlane bundle. If `SKIP_SETUP` is not set, [setup](https://github.com/mattermost/mattermost-mobile/blob/main/scripts/build.sh#L40-L81) will all do it.

---

Thanks sounds good to me.